src/crypto/store/memory-crypto-store.ts

```diff
@@ -292,17 +292,21 @@ export class MemoryCryptoStore implements CryptoStore {
         func(this.crossSigningKeys);
     }
 
-    public getSecretStorePrivateKey(txn: unknown, func: (key: IEncryptedPayload | null) => void, type: string): void {
-        const result = this.privateKeys[type];
+    public getSecretStorePrivateKey<T = IEncryptedPayload>(
```

Comment:
Are these type parameters meant to be `extends IEncryptedPayload`, or can they really be anything?

Reply:
They really can be anything, its entirely opaque object storage

Reply:
I have improved this using a mapped type, so now everything we store has to be typed better